packages/sources/dar/src/transport/price.ts

```diff
@@ -34,16 +39,23 @@ const hitTrackingLevels: { [level: string]: boolean } = {
   trace: true,
 }
 
-export const transport = new WebSocketTransport<WsTransportTypes>({
-  url: (context) => context.adapterSettings.WS_API_ENDPOINT,
-  options: async (context) => {
+export const config: WebSocketTransportConfig<WsTransportTypes> = {
+  url: (context: EndpointContext<WsTransportTypes>) => context.adapterSettings.WS_API_ENDPOINT,
+  options: async (context, desiredSubs) => {
     const token = await getAuthToken(context.adapterSettings)
+
+    const subscribeAssets: string[] = []
+    for (const pair in desiredSubs) {
+      subscribeAssets.push(desiredSubs[pair]['base'].toLowerCase())
+    }
+    const subscribe_assets = subscribeAssets.join(',')
```

Comment:
Replacing this to use `map` would avoid needing the slightly confusing `subscribeAssets` and `subscribe_assets`:

```suggestion
    const subscribeAssets = desiredSubs.map((sub) => sub.pair.base.toLowerCase()).join(',')
```